presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -4760,6 +4773,44 @@ public void testExecuteWithParametersInGroupBy()
         }
     }
 
+    @Test
+    public void testExecuteUsingSubqueryFails()
+    {
+        try {
+            String query = "SELECT ?";
+            Session session = Session.builder(getSession())
+                    .addPreparedStatement("my_query", query)
+                    .build();
+            computeActual(session, "EXECUTE my_query USING (SELECT 1 from nation)");
+            fail("nonLiteral parameters should fail");
+        }
+        catch (SemanticException e) {
+            assertEquals(e.getCode(), EXPRESSION_NOT_CONSTANT);
+        }
+        catch (RuntimeException e) {
+            assertEquals(e.getMessage(), "line 1:24: Constant expression cannot contain table references");
+        }
+    }
+
+    @Test
+    public void testExecuteUsingColumnReferenceFails()
+    {
+        try {
+            String query = "SELECT ? from nation";
+            Session session = Session.builder(getSession())
+                    .addPreparedStatement("my_query", query)
+                    .build();
+            computeActual(session, "EXECUTE my_query USING \"nationkey\"");
+            fail("nonLiteral parameters should fail");
+        }
+        catch (SemanticException e) {
+            assertEquals(e.getCode(), EXPRESSION_NOT_CONSTANT);
+        }
+        catch (RuntimeException e) {
+            assertEquals(e.getMessage(), "line 1:24: Constant expression cannot contain column references");
```

Comment:
Also add a test for passing * - AllColumns I assume that you just added a visitor above.